### PR TITLE
fix(terraform): fix valid value for aws keyspaces_table encryption_specification type

### LINKED ISSUE #5744: CKV_AWS_265 false positive
**Describe the issue**
CKV_AWS_265 fails checks on KeySpaces not using a Customer Managed KMS key even though it is configured to do so

This is because the code is looking for ["CUSTOMER_MANAGED_KEY"](https://github.com/bridgecrewio/checkov/blob/main/checkov/terraform/checks/resource/aws/KeyspacesTableUsesCMK.py#L17)

When the specification is listed as actually ["CUSTOMER_MANAGED_KMS_KEY"](https://docs.aws.amazon.com/keyspaces/latest/APIReference/API_EncryptionSpecification.html)

**Examples**
```
resource "aws_keyspaces_table" "keyspace_table" {
  keyspace_name = "foo"
  table_name    = "bar"
  encryption_specification {
   ## This should not fail the check CKV_AWS_265 but it does
    kms_key_identifier = var.kms_key_arn
    type               = "CUSTOMER_MANAGED_KMS_KEY"
  }
}
```

**Version (please complete the following information):**
 - 3.0.32

**Additional context**

```
~/Downloads> checkov -f ./keyspaces.tf --support
2023-11-10 09:21:38,953 [MainThread  ] [WARNI]  --bc-api-key argument is required when using --support
[ terraform framework ]: 100%|████████████████████|[1/1], Current File Scanned=keyspaces.tf
[ secrets framework ]: 100%|████████████████████|[1/1], Current File Scanned=./keyspaces.tf

       _               _
   ___| |__   ___  ___| | _______   __
  / __| '_ \ / _ \/ __| |/ / _ \ \ / /
 | (__| | | |  __/ (__|   < (_) \ V /
  \___|_| |_|\___|\___|_|\_\___/ \_/

By Prisma Cloud | version: 3.0.32

terraform scan results:

Passed checks: 0, Failed checks: 1, Skipped checks: 0

Check: CKV_AWS_265: "Ensure Keyspaces Table uses CMK"
	FAILED for resource: aws_keyspaces_table.keyspace_table
	File: /keyspaces.tf:1-9
	Guide: https://docs.prismacloud.io/en/enterprise-edition/policy-reference/aws-policies/aws-general-policies/ensure-aws-keyspace-table-uses-customer-managed-keys-cmks

		1 | resource "aws_keyspaces_table" "keyspace_table" {
		2 |   keyspace_name = "foo"
		3 |   table_name    = "bar"
		4 |   encryption_specification {
		5 |    ## This should not fail the check CKV_AWS_265 but it does
		6 |     kms_key_identifier = var.kms_key_arn
		7 |     type               = "CUSTOMER_MANAGED_KMS_KEY"
		8 |   }
		9 | }
```

### PR DESCRIPTION
**Pull request by tal66**
hi, this closes #5744.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Saarett** [comment]
Hey @tal66 , I think it could be either `AWS_OWNED_KMS_KEY` and `CUSTOMER_MANAGED_KMS_KEY`, what do you think?
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/keyspaces_table#type

**Comment by tal66** [comment]
where is it different in my commit? in the __doc__ i mention exactly these values, and check for the recommended CUSTOMER_MANAGED_KMS_KEY

File: checkov/terraform/checks/resource/aws/KeyspacesTableUsesCMK.py [hunk 1]
```diff
@@ -14,7 +18,7 @@ def scan_resource_conf(self, conf) -> CheckResult:
         if conf.get("encryption_specification") and isinstance(conf.get("encryption_specification"), list):
             encrypt = conf.get("encryption_specification")[0]
             if encrypt.get("kms_key_identifier") and isinstance(encrypt.get("kms_key_identifier"), list):
-                if encrypt.get("type") == ["CUSTOMER_MANAGED_KEY"]:
+                if encrypt.get("type") == ["CUSTOMER_MANAGED_KMS_KEY"]:
```
Comment (Saarett) [inline checkov/terraform/checks/resource/aws/KeyspacesTableUsesCMK.py:21]:
```suggestion
                if encrypt.get("type") == ["CUSTOMER_MANAGED_KMS_KEY"] or encrypt.get("type") == ["CUSTOMER_MANAGED_KEY"]:
```

@tal66 So what I meant is this

Comment (tal66) [reply checkov/terraform/checks/resource/aws/KeyspacesTableUsesCMK.py:21]: but the aws doc linked in the issue says it's not valid, why keep both?

Comment (gruebel) [reply checkov/terraform/checks/resource/aws/KeyspacesTableUsesCMK.py:21]: chipping in 🙂 it is `CUSTOMER_MANAGED_KMS_KEY` indeed `CUSTOMER_MANAGED_KEY` doesn't exist as can seen in the official API docs https://docs.aws.amazon.com/keyspaces/latest/APIReference/API_CreateTable.html#keyspaces-CreateTable-request-encryptionSpecification

**Review by gruebel — APPROVED** [review]
thanks, looks good 🏅

File: checkov/terraform/checks/resource/aws/KeyspacesTableUsesCMK.py (hunk 1 above)
Comment (Saarett) [reply checkov/terraform/checks/resource/aws/KeyspacesTableUsesCMK.py:21]: My bad, I see it now. Thanks!

**Review by Saarett — APPROVED** [review]

**Pull request merged**